assets/base.css

```diff
@@ -1216,86 +1216,111 @@ deferred-media {
-  --shadow-vertical-offset: var(--buttons-shadow-vertical-offset);
-  --shadow-blur-radius: var(--buttons-shadow-blur-radius);
-  --shadow-opacity: var(--buttons-shadow-opacity);
+.button:after,
+.shopify-challenge__button:after,
+.customer button:after,
+.shopify-payment-button__button--unbranded:after {
+  content: '';
+  position: absolute;
+  inset: var(--buttons-border-width);
+  z-index: 1;
   border-radius: var(--buttons-radius);
-  filter: drop-shadow(
-    var(--shadow-horizontal-offset)
-    var(--shadow-vertical-offset)
-    var(--shadow-blur-radius)
-    rgba(var(--color-button), var(--shadow-opacity))
-  );
+  box-shadow: 0 0 0 calc(var(--buttons-border-width) + var(--border-offset)) rgba(var(--color-button-text), var(--border-opacity)),
+    0 0 0 var(--buttons-border-width) rgba(var(--color-button), 1);
   transition: box-shadow var(--duration-short) ease;
 }
 
-.button--secondary {
-  box-shadow: 0 0 0 var(--buttons-border-width) rgba(var(--color-button-text), var(--buttons-border-opacity));
-  filter: drop-shadow(
-    var(--shadow-horizontal-offset)
-    var(--shadow-vertical-offset)
-    var(--shadow-blur-radius)
-    rgba(var(--color-button-text), var(--shadow-opacity))
-  );
+.button:not([disabled]):hover::after,
+.shopify-challenge__button:hover::after,
+.customer button:hover::after,
+.shopify-payment-button__button--unbranded:hover::after {
+  --border-offset: 1.3px;
+  box-shadow: 0 0 0 calc(var(--buttons-border-width) + var(--border-offset)) rgba(var(--color-button-text), var(--border-opacity)),
+    0 0 0 calc(var(--buttons-border-width) + 1px) rgba(var(--color-button), 1);
 }
 
-.button:focus-visible,
-.shopify-payment-button__button--unbranded:focus-visible,
-.shopify-payment-button [role="button"]:focus-visible {
-  box-shadow: 0 0 0 0.1rem rgba(var(--color-button), var(--alpha-button-border)),
-    0 0 0 0.3rem rgb(var(--color-background)),
-    0 0 0.5rem 0.4rem rgba(var(--color-foreground), 0.3);
+.button--secondary:after {
+  --border-opacity: var(--buttons-border-opacity);
 }
 
+.button:focus-visible,
 .button:focus,
+.button.focused,
+.shopify-payment-button__button--unbranded:focus-visible,
+.shopify-payment-button [role="button"]:focus-visible,
 .shopify-payment-button__button--unbranded:focus,
 .shopify-payment-button [role="button"]:focus {
-  box-shadow: 0 0 0 0.1rem rgba(var(--color-button), var(--alpha-button-border)),
-    0 0 0 0.3rem rgb(var(--color-background)),
+  outline: 0;
+  box-shadow: 0 0 0 0.3rem rgb(var(--color-background)),
+    0 0 0 0.5rem rgba(var(--color-foreground), 0.5),
     0 0 0.5rem 0.4rem rgba(var(--color-foreground), 0.3);
 }
 
-.button:focus:not(:focus-visible),
-.shopify-payment-button__button--unbranded:focus:not(:focus-visible),
-.shopify-payment-button [role="button"]:focus:not(:focus-visible) {
-  box-shadow: 0 0 0 0.1rem rgba(var(--color-button), var(--alpha-button-border));
+.button:focus:not(:focus-visible, .focused),
```

Comment:
I think here we need to do `button:focus:not(:focus-visible):not(.focused)` instead as I remember we had to tweak it in the past as it was messing something. By the looks of it, IE. Not sure if that was the only reason but we wanted to not break atc buttons on IE even though it's not really supported.